### ISSUE TITLE
[tracking] Replace fsouza/go-dockerclient with docker/docker/client

### ISSUE THREAD

**Issue by ingvagabund** [openshift/oc#352]
Placeholder for tracking the issue

The latest changes in openshift/library-go pulled changes in golang.org/x/sys which are incompatible with the current revision of github.com/docker/docker. In order to have the docker built properly again, https://github.com/moby/moby/commit/c3a0a3744636069f43197eb18245aaae89f568e5 needs to be present. However, updating docker to the revision makes fsouza/go-dockerclient incompatible. Also, k8s.io/kubernetes is already fsouza/go-dockerclient code free, we should do the same for oc (and other openshift packages).

Info:
- docker.Client required to have `Ping() error` method (new method has `Ping(ctx context.Context) (types.Ping, error)` signature)
- `ListImages(opts docker.ListImagesOptions) ([]docker.APIImages, error)` replaced with `ImageList(ctx context.Context, options types.ImageListOptions) ([]types.ImageSummary, error)`
- `InspectImage(name string) (*docker.Image, error)` replaced with `ImageInspectWithRaw(ctx context.Context, imageID string) (types.ImageInspect, []byte, error)`
- `github.com.fsouza/go-dockerclient.Port` replaced with `github.com/docker/go-connections/nat.Port`

DockerConfig field without an equivalent field in container.Config:
```
Memory            int64               `json:"Memory,omitempty" yaml:"Memory,omitempty"`
MemorySwap        int64               `json:"MemorySwap,omitempty" yaml:"MemorySwap,omitempty"`
MemoryReservation int64               `json:"MemoryReservation,omitempty" yaml:"MemoryReservation,omitempty"`
KernelMemory      int64               `json:"KernelMemory,omitempty" yaml:"KernelMemory,omitempty"`
PidsLimit         int64               `json:"PidsLimit,omitempty" yaml:"PidsLimit,omitempty"`
CPUShares         int64               `json:"CpuShares,omitempty" yaml:"CpuShares,omitempty"`
CPUSet            string              `json:"Cpuset,omitempty" yaml:"Cpuset,omitempty"`
PortSpecs         []string            `json:"PortSpecs,omitempty" yaml:"PortSpecs,omitempty"`
DNS               []string            `json:"Dns,omitempty" yaml:"Dns,omitempty"` // For Docker API v1.9 and below only
VolumeDriver      string              `json:"VolumeDriver,omitempty" yaml:"VolumeDriver,omitempty"`
VolumesFrom       string              `json:"VolumesFrom,omitempty" yaml:"VolumesFrom,omitempty"`
SecurityOpts    []string            `json:"SecurityOpts,omitempty"`
Mounts            []Mount             `json:"Mounts,omitempty" yaml:"Mounts,omitempty"`
```

**Comment by ingvagabund** [comment]
So I believe I found the right commit: https://github.com/moby/moby/commit/d1d5f6476656c6aad457e2a91d3436e66b6f2251. However, it's a commit that was merged into the repo. It should be part of #40021. Though, when you open the PR there are only two commits:
c3a0a3744636069f43197eb18245aaae89f568e5
4e0836f01b5f74237c7b8467af67988add61e247

Originally, I used c3a0a3744636069f43197eb18245aaae89f568e5 (as anyone would). However, when you diff both commits:
```
$ git diff d1d5f6476656 c3a0a3744636069f43197eb18245aaae89f568e5 pkg/homedir/
diff --git a/pkg/homedir/homedir_unix.go b/pkg/homedir/homedir_unix.go
index 441bd72..dcdce40 100644
--- a/pkg/homedir/homedir_unix.go
+++ b/pkg/homedir/homedir_unix.go
@@ -1,4 +1,4 @@
-// +build !windows
+// +build !windows,cgo !windows,osusergo
 
 package homedir // import "github.com/docker/docker/pkg/homedir"
 
@@ -16,11 +16,8 @@ func Key() string {
 // Get returns the home directory of the current user with the help of
 // environment variables depending on the target operating system.
 // Returned path should be used with "path/filepath" to form new paths.
-//
-// If linking statically with cgo enabled against glibc, ensure the
-// osusergo build tag is used.
-//
-// If needing to do nss lookups, do not disable cgo or set osusergo.
+// If compiling statically, ensure the osusergo build tag is used.
+// If needing to do nss lookups, do not compile statically.
 func Get() string {
        home := os.Getenv(Key())
        if home == "" {
```
Basically, the +build directive changed and it made the difference. Making Get() not available for darwin. Thus, failing the build at the first place. However, when I change the commit from c3a0a3744636069f43197eb18245aaae89f568e5 to d1d5f6476656c6aad457e2a91d3436e66b6f2251, I get a clean rpm build and don't have to replace fsouza/go-dockerclient with docker/docker.
